### PR TITLE
fix(provider): remove waitgroup for lease withdraw

### PR DESCRIPTION
**Pull request by troian**
fixes #eng-282

Signed-off-by: Artur Troian <troian.ap@gmail.com>

### CODE REVIEW

File: provider/cluster/manager.go [hunk 1]
```diff
@@ -239,13 +239,14 @@ loop:
 		dm.log.Debug("read from runch during shutdown")
 	}
 
-	dm.log.Debug("waiting on dm.wg")
-	dm.wg.Wait()
-
 	if nil != dm.withdrawal {
 		dm.log.Debug("waiting on withdrawal")
-		dm.withdrawal.lc.Shutdown(nil)
+		dm.withdrawal.lc.Done()
```
Comment (tidrolpolelsef) [inline provider/cluster/manager.go:244]: Doesn't this return a channel? What 's the point of calling it here if we don't use the channel for something?

Comment (troian) [reply provider/cluster/manager.go:244]: good catch!

Comment (boz) [reply provider/cluster/manager.go:244]: you can use `ShutdownAsync()` if you don't want to wait.  `Done()` does just return a channel that is readable when the object is done.

Comment (boz) [reply provider/cluster/manager.go:244]: ideally the `withdrawal` object would have `Shutdown()`, `Done()` methods & you can wait on `<-dm.withdrawal.Done()`.

Comment (troian) [reply provider/cluster/manager.go:244]: it supposed to be `<-dm.withdrawal.lc.Done()`. i just missed read part

File: .github/workflows/k8s-integration.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ on:
   pull_request:
     paths:
       - pkg/**
-      - provider/cluster/kube/*
+      - provider/*
```
[the remark this answers is not in the document]
Comment (tidrolpolelsef) [reply .github/workflows/k8s-integration.yaml:10]: Yeah make that a separate PR please.

**Review by tidrolpolelsef — APPROVED** [review]

**Pull request merged**